From e3fd6506b5de2601f28463f63fe0b207420460c6 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 12 Aug 2013 15:13:15 +0300
Subject: [PORT FROM MAIN] Telephony: Fix issue in showing network information with data only sim

BZ: 127199

With data only SIM cards, device shows no service but
browsing works.

When the device is voice capable, ServiceState always
returns the circuit-switched network registration state.
So, when the device is not registered to circuit-switched
network service state is shown as "No Service" and the
siganl bars are also empty.

When the device is not registered to circuit-switched
network but attached to data network then update the
service state with the data registration state.

Severity: High

Change-Id: Ib43a49f89e61b86c6549ad3b15df43965097b187
Orig-Change-Id: I6c7c90cc374214aa1954b7ed9cf391afadad7ecc
Category: device enablement
Domain: Telephony-Modem-Common
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/gsm/GsmServiceStateTracker.java      |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
index 3e988f8..8f300e4 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
@@ -728,6 +728,13 @@ final class GsmServiceStateTracker extends ServiceStateTracker {
             loge("Exception while polling service state. Probably malformed RIL response." + ex);
         }
 
+        final boolean voice_capable = mPhone.getContext().getResources().getBoolean(
+                com.android.internal.R.bool.config_voice_capable);
+        if ((!voice_capable || mNewSS.getState() != ServiceState.STATE_IN_SERVICE)
+                && mNewSS.getDataRegState() == ServiceState.STATE_IN_SERVICE) {
+             mNewSS.setState(mNewSS.getDataRegState());
+        }
+
         mPollingContext[0]--;
 
         if (mPollingContext[0] == 0) {
-- 
1.7.4.1

